### DISCUSSION TITLE
How to add localization in the custom uninstall dialog box

### DISCUSSION THREAD

**Discussion by Vaibhav1825** [oleg-shilo/wixsharp#1716] [Q&A]
Hello @oleg-shilo, I want to implement localization in the custom uninstall dialog box of my project.
I have made the custom dialog for the uninstall and called that dialog on after uninstall but I am unable to do localization of that uninstall custom dialog.
For adding localization in the other custom dialog boxes, I have used the **.LocalizeWith(this.Runtime.Localize)** feature, but in the uninstall dialog I am unable to use the session property or access the session.
Can you please help me on this about how to implement localization in the custom uninstall dialog.

**Comment by oleg-shilo** [comment]
You are right. the session object is no longer available after the session is terminated.
What you can do is extract the localization information earlier, before the session is over and store it for later use.

IE read the localization file from your msi in some early dialogs:
```c#
// project
var project = new ManagedProject(...
                  new Binary("myLang.wxl"),

// uninstall dialog
class MyUninstallDialog       
{
    public static Func<string, string>Localize;
    . . .
    this.LocalizeWith(this.Localize)
                  
// early dialog  (e.g. WelcomeDlg)
var bytes = sessiom.ReadBinary("myLang.wxl"); // alternatively you can save the localization data to the temp file to read it later
var translations = new ResourcesData();
translations.InitFromWxl(bytes); 
MyUninstallDialog.Localize =  text => translations.ContainsKey(text) ? translations.[text] : text;
   
```
